### PR TITLE
MODNOTES-93 - Improvements: Modify metadata triggers not to use addit…

### PR DESCRIPTION
**Pull request by natali-zaitseva**
## Purpose
Updates for https://issues.folio.org/browse/MODNOTES-93  

## Approach
* modified scripts not to use additional columns
* removed columns created_by and creation_date for note tables
* modified tests to adjust the updated logic

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/NoteTypesImplTest.java [hunk 1]
```diff
@@ -448,7 +447,7 @@ public void shouldReturn404OnPutWhenNoteNotFound() throws IOException, URISyntax
   }
 
   @Test
-  public void shouldReturn400OnPutWhenRequestIsInvalid() {
+  public void shouldReturn422OnPutWhenRequestIsInvalid() {
```
Comment (sduvvuri-ebsco) [inline src/test/java/org/folio/rest/impl/NoteTypesImplTest.java]: Just a quick sanity check here -> why did we decide on a 400 before and now a 422?

File: src/test/java/org/folio/rest/impl/NoteTypesImplTest.java [hunk 2]
```diff
@@ -536,7 +534,7 @@ public void shouldReturn400WhenUserIdIsMissing() {
   }
 
   @Test
-  public void shouldReturn401WhenIncorrectTenant() {
+  public void shouldReturn400WhenIncorrectTenant() {
```
Comment (sduvvuri-ebsco) [inline src/test/java/org/folio/rest/impl/NoteTypesImplTest.java]: Similarly, when incorrect tenant is given, why did we use unauthorized before and now a bad request?

File: src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql [hunk 3]
```diff
@@ -30,3 +34,9 @@ language 'plpgsql';
 DROP TRIGGER IF EXISTS set_note_data_md_json_trigger ON note_data CASCADE;
 
 CREATE TRIGGER set_note_data_md_json_trigger BEFORE UPDATE ON note_data   FOR EACH ROW EXECUTE PROCEDURE set_note_data_md_json();
+
+DROP TRIGGER IF EXISTS set_note_data_md_trigger ON note_data CASCADE;
```
Comment (sduvvuri-ebsco) [inline src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql]: Jus a quick sanity check here -> we are dropping and creating "set_note_data_md_json_trigger". Is "set_note_data_md_trigger" another trigger defined elsewhere?

File: src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql [hunk 4]
```diff
@@ -30,3 +34,9 @@ language 'plpgsql';
...
+
+DROP FUNCTION IF EXISTS note_data_set_md();
```
Comment (sduvvuri-ebsco) [inline src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql]: Similarly, "set_note_data_md_json()" is the function I see defined. Where is "note_data_set_md()"?

File: src/main/resources/templates/db_scripts/note_type_update_metadata_user_info.sql [hunk 5]
```diff
@@ -31,5 +35,8 @@ DROP TRIGGER IF EXISTS set_note_type_md_json_trigger ON note_type CASCADE;
 
 CREATE TRIGGER set_note_type_md_json_trigger BEFORE UPDATE ON note_type   FOR EACH ROW EXECUTE PROCEDURE set_note_type_md_json();
 
+DROP TRIGGER IF EXISTS set_note_type_md_trigger ON note_type CASCADE;
 
+DROP FUNCTION IF EXISTS note_type_set_md();
```
Comment (sduvvuri-ebsco) [inline src/main/resources/templates/db_scripts/note_type_update_metadata_user_info.sql]: similar question as above here for both trigger and function. Thanks!

File: src/test/java/org/folio/rest/impl/NoteTypesImplTest.java (hunk 1 above)
Comment (natali-zaitseva) [reply src/test/java/org/folio/rest/impl/NoteTypesImplTest.java]: @sduvvuri-ebsco just changed the name of the method to conform with response code

File: src/test/java/org/folio/rest/impl/NoteTypesImplTest.java (hunk 2 above)
Comment (natali-zaitseva) [reply src/test/java/org/folio/rest/impl/NoteTypesImplTest.java]: @sduvvuri-ebsco thanks, fixed

File: src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql (hunk 3 above)
Comment (natali-zaitseva) [reply src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql]: @sduvvuri-ebsco `set_note_data_md_trigger`, as well as other triggers, are created by Postgres Client. In the logs, you can find the information about triggers/function creation.

Comment (natali-zaitseva) [reply src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql]: The sample code for trigger:
Screen Shot 2019-05-06 at 2 41 38 PM

File: src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql (hunk 4 above)
Comment (natali-zaitseva) [reply src/main/resources/templates/db_scripts/note_data_update_metadata_user_info.sql]: @sduvvuri-ebsco we do not need  `note_data_set_md()` function, that is why we just delete it. The main reason of it was to store the creation date as a separate column.

**Pull request merged**